packages/realm/bindgen/src/templates/typescript.ts

```diff
@@ -162,31 +162,21 @@ export function generate({ spec: rawSpec, file }: TemplateContext): void {
 
   const spec = doJsPasses(bindModel(rawSpec));
 
-  const coreOut = file("core.ts", "eslint", "typescript-checker");
```

Comment:
This doesn't make sense. core.ts was a plain ts file that was `export * from "./core"` in both the wrapper and in the .d.ts file. It doesn't need to export anything from "realm/binding" because the module imported by that name will transitively re-export all of its contents in both TS declarations, and in JS runtime land.

Also, you totally can do module augmentation from a ts file. I do it in [bindgen](https://github.com/realm/realm-js/blob/main/packages/realm/bindgen/src/js-passes.ts#L95-L114). Also [like this](https://github.com/realm/realm-js/blob/main/packages/realm/bindgen/src/templates/node.ts#L586-L594)

Reply:
When I include `export * from "./core";` in the `declare module "realm/binding" {` block of the `native.d.mts` I get:

> Import or export declaration in an ambient module declaration cannot reference module through relative module name.ts(2439)

I'll try doing module augmentation from the `core.ts` again 🤔

Reply:
> Also, you totally can do module augmentation from a ts file. I do it in [bindgen](https://github.com/realm/realm-js/blob/main/packages/realm/bindgen/src/js-passes.ts#L95-L114). Also [like this](https://github.com/realm/realm-js/blob/main/packages/realm/bindgen/src/templates/node.ts#L586-L594)

Those are examples of augmentation of the module declaration, not the implementation. We'd need the latter to implement `core.ts` since it provides the implementation too?